### ISSUE TITLE
New Check: CompactSourceFileCheck to detect JEP 512 compact source files

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#20229]
Parent Issue #19971

## Description

Add a new check `CompactSourceFileCheck` that detects [JEP 512](https://openjdk.org/jeps/512) compact source files (top-level methods and fields with no enclosing class declaration) and fires a single violation on each such file. The check allows teams that require explicit class declarations to prohibit compact source files in their codebases.

## Rationale

Java 25 finalizes compact source files. A `.java` file can contain top-level methods and fields, and the compiler wraps them in an implicit class that:

- has no name (cannot be referred to from other files by any qualified path),
- is restricted to the unnamed package (a `package …;` declaration is forbidden by JLS),
- is always `final` (cannot be subclassed),
- limits the file to a single implicit class.

This form is convenient for scripts, learning code, and single-file programs, but teams maintaining libraries, frameworks, or production code may want to forbid it entirely. There is no existing check that detects it.

## Proposed Design

**Token:** `TokenTypes.COMPACT_COMPILATION_UNIT` (required + default + acceptable).

Under the grammar implemented in #17810, `COMPACT_COMPILATION_UNIT` replaces `COMPILATION_UNIT` as the AST root **iff** the file is a compact source file. The check therefore registers for `COMPACT_COMPILATION_UNIT` and fires once when the root is visited. 

**Properties:** none, the check is binary by design: a file is either a compact source file or it isn't. 

**Violation key and message:**

| Key | Message |
|---|---|
| `compact.source.file` | `Compact source file is not allowed. Please use an explicit class declaration.` |

## Examples

**Configuration:**

```xml
<module name="Checker">
  <module name="TreeWalker">
    <module name="CompactSourceFile"/>
  </module>
</module>
```

**Compact source file (violation expected):**

```java
// Example.java
int counter = 3;

void main() {
    System.out.println(counter);
}
```

Expected output:

```
Example.java:1: Compact source file is not allowed. Please use an explicit class declaration. [CompactSourceFile]
```

**Ordinary class (no violation expected):**

```java
// Example.java
public class Example {
    static int counter = 3;

    public static void main(String[] args) {
        System.out.println(counter);
    }
}
```

Expected output: clean, no violations.

**Comment by romani** [comment]
I think this can be easier to be archived by xpathmatch instance that checks root node.

Usage of it will be very special. I pretty sure people will start using compact sources as scripts, instead of other languages scripts in their repositories. So full forbidition will be unlikely used.  If class can not be referenced, it awesome safety, nobody can misuse it, it will be used only launcher/script.

**Comment by mahfouz72** [comment]
Can be achieved by [IllegalToken](https://checkstyle.sourceforge.io/checks/coding/illegaltoken.html) also
```XML
<module name="IllegalToken">
   <property name="tokens" value="COMPACT_COMPILATION_UNIT"/>
</module>
```

**Comment by romani** [comment]
Elegant ∆∆∆

**Comment by romani** [comment]
@vivek-0509 , I think we can close issue. There is a way to archive prevention of compact sources, is user wants.

**Comment by vivek-0509** [comment]
@romani  Sure closing it
